### ISSUE TITLE
Set CMDSTAN environment variable to gloabl .cmdstanr directory rather than version

### ISSUE THREAD

**Issue by bwiernik** [stan-dev/cmdstanr#638]
**Is your feature request related to a problem? Please describe.**
I have cmdstan installed to a non-default directory (I'm on Windows and the default location is in a cloud storage folder). This means that I have to change my .Renviron every time I install a new version of cmdstan, rather than having the latest version automatically detected.

**Describe the solution you'd like**
I'd like to be able to set my CMDSTAN variable in .Renviron to be the `.cmdstanr` directory, rather than specific version directories. That way, the default when I call cmdstan could be to use the latest version in the directory.

To accomplish this, if the CMDSTAN path doesn't contain the CmdStan makefile, the logic from `cmdstanr::cmdstan_default_path()` would be run in the directory to find the latest version of cmdstan there.

**Comment by rok-cesnovar** [comment]
Thank you for the feature request Brenton. Seems like a valid use case and a simple fix. Hopefully, I will get to it this week.

**Comment by jgabry** [comment]
Yeah this sounds reasonable to me too.

**Comment by rok-cesnovar** [comment]
Should be addressed in #643 : 

Run below to test:
```r
remotes::install_github("stan-dev/cmdstanr@cmdstan-variable-refactor")
Sys.setenv("CMDSTAN" = "/top/level/folder/of/cmdstan/installs") # or set it in .Renviron

library(cmdstanr)
```
